src/functions.php

```diff
@@ -216,5 +217,9 @@ function recursive_copy($element) {
         return $element;
     }
 
+    if ( ! (new ReflectionClass($element))->isCloneable()) {
```

Comment:
Is cloneable is reported to not work with internal classes (in a note on http://php.net/manual/en/reflectionclass.iscloneable.php) — could that be a problem for us?

Reply:
That note was the reason I added `testCloneSupportsBSONTypes()`. I didn't look into `isCloneable()`, but it certainly appears to do the job for our BSON classes in PHP 5.5+.